### PR TITLE
Moving all cmd opts to json

### PR DESCRIPTION
**Pull request by teseoch**
- log stuff (level, path, is quiet)
- Num threads
- Output path
- Enable overwrite solver

### CODE REVIEW

File: src/polyfem/state/StateInit.cpp [hunk 1]
```diff
@@ -203,6 +216,15 @@ namespace polyfem
 
 		this->args = jse.inject_defaults(args_in, rules);
 
+		std::string out_path_log = this->args["output"]["log"]["path"];
```
Comment (teseoch) [inline src/polyfem/state/StateInit.cpp:219]: @zfergus this line doesn't work with strings, do you know why?

**Comment by teseoch** [comment]
I also added ignore case and ignore underscore to cli11, do we like it?

File: src/polyfem/state/StateInit.cpp [hunk 2]
```diff
@@ -179,31 +186,42 @@ namespace polyfem
...
+		if (!out_path_log.empty())
 		{
-			logger().error("invalid input json:\n{}", jse.log2str());
-			throw std::runtime_error("Invald input json file");
+			out_path_log = resolve_output_path(out_path_log);
 		}
-		// end of check
 
-		this->args = jse.inject_defaults(args_in, rules);
+		spdlog::level::level_enum log_level = this->args["output"]["log"]["level"];
```
Comment (zfergus) [inline src/polyfem/state/StateInit.cpp:220]: I dont know if this will work with ints, it should work with strings

**Review by zfergus — APPROVED** [review]

File: src/polyfem/state/StateInit.cpp (hunk 1 above)
Comment (zfergus) [reply src/polyfem/state/StateInit.cpp:214]: no, this looks like it would work

**Pull request merged**